Nodejs/Product/TestAdapter/TestExecutorWorker.cs

```diff
@@ -0,0 +1,392 @@
+                // Extract test from list of tests
+                var tests = this.currentTests.Where(n => n.DisplayName == testEvent.title);
+                if (tests.Any())
+                {
+                    switch (testEvent.type)
+                    {
+                        case "test start":
+                            {
+                                this.currentResult = new TestResult(tests.First())
+                                {
+                                    StartTime = DateTimeOffset.Now
+                                };
+                                this.frameworkHandle.RecordStart(tests.First());
+                            }
+                            break;
+                        case "result":
+                            {
+                                RecordEnd(tests.First(), testEvent.result);
+                            }
+                            break;
+                        case "pending":
+                            {
+                                this.currentResult = new TestResult(tests.First());
+                                RecordEnd(tests.First(), testEvent.result);
+                            }
+                            break;
+                    }
+                }
+                else if (testEvent.type == "suite end")
+                {
+                    this.currentResultObject = testEvent.result;
+                    this.testsCompleted.Set();
+                }
+            }
+            catch (JsonReaderException)
+            {
+                // Often lines emitted while running tests are not test results, and thus will fail to parse above
+            }
+
+            void RecordEnd(TestCase test, ResultObject resultObject)
+            {
+                var standardOutputLines = resultObject.stdout.Split('\n');
+                var standardErrorLines = resultObject.stderr.Split('\n');
+
+                if (resultObject.pending == true)
+                {
+                    this.currentResult.Outcome = TestOutcome.Skipped;
+                }
+                else
+                {
+                    this.currentResult.EndTime = DateTimeOffset.Now;
+                    this.currentResult.Duration = this.currentResult.EndTime - this.currentResult.StartTime;
+                    this.currentResult.Outcome = resultObject.passed ? TestOutcome.Passed : TestOutcome.Failed;
+                }
+
+                this.currentResult.Messages.Add(new TestResultMessage(TestResultMessage.StandardOutCategory, string.Join(Environment.NewLine, standardOutputLines)));
+                this.currentResult.Messages.Add(new TestResultMessage(TestResultMessage.StandardErrorCategory, string.Join(Environment.NewLine, standardErrorLines)));
+                this.currentResult.Messages.Add(new TestResultMessage(TestResultMessage.AdditionalInfoCategory, string.Join(Environment.NewLine, standardErrorLines)));
```

Comment:
If writing the same thing to two different streams is intentional, consider extracting a local.